src/targets/gpu/gemm_impl.cpp

```diff
@@ -123,13 +123,13 @@ auto rocblas_invoke(F f, Pack p, Ts... xs)
 
 static bool is_transposed(const shape& s) { return s.transposed() and s.strides().back() != 1; }
 
-static rocblas_int get_batch_stride(const argument& a)
+static rocblas_int get_batch_stride(const shape& s)
```

Comment:
Did you say you were going to make this explicit?

Reply:
Ah, I see it now.  A different PR.  Which will have to be merged into this one to make it work right.

Reply:
The explicit constructor is not needed for this PR. I only added so we can catch these mistakes on compile-time in the future.